dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/sam/SamPermissionsImpl.java

```diff
@@ -0,0 +1,257 @@
+package io.dockstore.webservice.permissions.sam;
+
+import java.text.MessageFormat;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.dockstore.webservice.CustomWebApplicationException;
+import io.dockstore.webservice.DockstoreWebserviceConfiguration;
+import io.dockstore.webservice.core.Token;
+import io.dockstore.webservice.core.TokenType;
+import io.dockstore.webservice.core.User;
+import io.dockstore.webservice.core.Workflow;
+import io.dockstore.webservice.helpers.GoogleHelper;
+import io.dockstore.webservice.jdbi.TokenDAO;
+import io.dockstore.webservice.permissions.Action;
+import io.dockstore.webservice.permissions.Permission;
+import io.dockstore.webservice.permissions.PermissionsInterface;
+import io.dockstore.webservice.permissions.Role;
+import io.swagger.sam.client.ApiClient;
+import io.swagger.sam.client.ApiException;
+import io.swagger.sam.client.JSON;
+import io.swagger.sam.client.api.ResourcesApi;
+import io.swagger.sam.client.model.AccessPolicyMembership;
+import io.swagger.sam.client.model.AccessPolicyResponseEntry;
+import io.swagger.sam.client.model.ErrorReport;
+import io.swagger.sam.client.model.ResourceAndAccessPolicy;
+import org.apache.http.HttpStatus;
+import org.glassfish.jersey.client.ClientConfig;
+import org.glassfish.jersey.client.ClientProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * An implementation of the {@link PermissionsInterface} that makes
+ * calls to SAM.
+ */
+public class SamPermissionsImpl implements PermissionsInterface {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SamPermissionsImpl.class);
+
+    private static Map<String, Role> samPermissionMap = new HashMap<>();
```

Comment:
Document maps when they use types like Strings, Pairs, etc. where the value isn't obvious

Reply:
Fixing in next commit